engine/paymail/paymail_service_client.go

```diff
@@ -0,0 +1,209 @@
+package paymail
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/bitcoin-sv/go-paymail"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/mrz1836/go-cachestore"
+)
+
+const cacheKeyCapabilities = "paymail-capabilities-"
+const cacheTTLCapabilities = 60 * time.Minute
+
+// PayloadFormat is the format of the paymail payload
+type PayloadFormat uint32
+
+// Types of Paymail payload formats
+const (
+	BasicPaymailPayloadFormat PayloadFormat = iota
+	BeefPaymailPayloadFormat
+)
+
+func (format PayloadFormat) String() string {
+	switch format {
+	case BasicPaymailPayloadFormat:
+		return "BasicPaymailPayloadFormat"
+
+	case BeefPaymailPayloadFormat:
+		return "BeefPaymailPayloadFormat"
+
+	default:
+		return fmt.Sprintf("%d", uint32(format))
+	}
+}
+
+// ServiceClient is a service that aims to make easier paymail operations.
+type ServiceClient interface {
+	GetSanitizedPaymail(addr string) (*paymail.SanitisedPaymail, error)
+	GetCapabilities(ctx context.Context, domain string) (*paymail.CapabilitiesPayload, error)
+	GetP2P(ctx context.Context, domain string) (success bool, p2pDestinationURL, p2pSubmitTxURL string, format PayloadFormat)
+	StartP2PTransaction(alias, domain, p2pDestinationURL string, satoshis uint64) (*paymail.PaymentDestinationPayload, error)
+	GetPkiForPaymail(ctx context.Context, sPaymail *paymail.SanitisedPaymail) (*paymail.PKIResponse, error)
+	AddContactRequest(ctx context.Context, receiverPaymail *paymail.SanitisedPaymail, contactData *paymail.PikeContactRequestPayload) (*paymail.PikeContactRequestResponse, error)
+}
+
+type service struct {
+	cs cachestore.ClientInterface
+	pc paymail.ClientInterface
+}
+
+// NewServiceClient creates a new paymail service client
+func NewServiceClient(cs cachestore.ClientInterface, pc paymail.ClientInterface) (ServiceClient, error) {
+	if pc == nil {
+		return nil, spverrors.Newf("paymail client is required")
```

Comment:
Can this happen on runtime - by design?
Or it will occur only by a developer mistake - in this case I would suggest - `panic`